### PR TITLE
Add buttons

### PR DESCRIPTION
**Pull request by maya**
This adds buttons to the library.

This is what it looks like:
screen shot 2015-07-01 at 4 33 39 pm

Here are the normal, active, and hover states, which are live:
active and hover

Thanks @noahmanger for helping with the focus state for rounded corners!

### CODE REVIEW

File: assets/_scss/elements/_buttons.scss [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+input[type="button"].button-big,
+input[type="submit"].button-big,
+input[type="reset"].button-big,
+input[type="image"].button-big,
+button[type="button"].button-big,
+button[type="submit"].button-big,
+button[type="reset"].button-big {
+  font-size: 1.9rem;
+  padding: 1.5rem 3rem;
```
Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: From your research, do you think people will want to control button size? We could potentially variablize this sizing stuff.

File: assets/_scss/elements/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+button[type="reset"] {
+  -webkit-appearance: none;
+  background-color: $color-grey-900;
+  border: 0;
+  border-radius: $usa-border-radius;
+  color: $color-white;
+  font-size: $base-font-size;
+  font-weight: $font-bold;  
+  line-height: 1;
+  text-align: center;
```
Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: Please keep properties alphabetically ordered according to style guide.

File: assets/_scss/elements/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  margin: {
+    top: .5em;
+    bottom: .5em;
+    right: .5em;
+  }
+  outline: none;
+  padding: 1rem 2rem;
+  -webkit-font-smoothing: antialiased;
```
Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: Please keep - properties together either at top or bottom.

File: assets/_scss/elements/_buttons.scss (hunk 2 above)
Comment (maya) [inline assets/_scss/elements/_buttons.scss]: Fixed in d2f6ac5

**Comment by msecret** [comment]
Future considerations:

- Allowing a `button` tag to be styled like a text link. This is helpful for when you're using JS for links and don't want to put an `href` attribute. So this would probably be `.unstyled` class on `button`.
- Being able to style a `<a>` tag like a button.

File: assets/_scss/elements/_buttons.scss [hunk 4]
```diff
@@ -0,0 +1,77 @@
+// Buttons
+
+input[type="button"],
+input[type="submit"],
+input[type="reset"],
+input[type="image"],
+button[type="button"],
+button[type="submit"],
+button[type="reset"] {
+  -webkit-appearance: none;
```
Comment (msecret) [inline assets/_scss/elements/_buttons.scss]: Remove all prefixes and replace with what would be in auto prefixer:

https://github.com/postcss/autoprefixer/wiki/support-list

File: assets/_scss/elements/_buttons.scss (hunk 3 above)
Comment (maya) [inline assets/_scss/elements/_buttons.scss]: Done in  d2f6ac5

**Comment by Scotchester** [comment]
:+1: to being able to stile `a`, `button`, and `input` elements using these button styles.

**Pull request merged**